ui/dashboards/src/components/PanelDrawer/PanelSpecEditor.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ErrorAlert, JSONEditor } from '@perses-dev/components';
+import { PanelDefinition, QueryDefinition, UnknownSpec } from '@perses-dev/core';
+import {
+  usePlugin,
+  PanelPlugin,
+  OptionsEditorTabsProps,
+  TimeSeriesQueryEditor,
+  OptionsEditorTabs,
+} from '@perses-dev/plugin-system';
+
+export interface PanelSpecEditorProps {
+  panelDefinition: PanelDefinition;
+  onQueriesChange: (queries: QueryDefinition[]) => void;
+  onPluginSpecChange: (spec: UnknownSpec) => void;
+  onJSONChange: (panelDefinition: PanelDefinition) => void;
+}
+
+export function PanelSpecEditor(props: PanelSpecEditorProps) {
+  const { panelDefinition, onJSONChange, onQueriesChange, onPluginSpecChange } = props;
+  const { kind } = panelDefinition.spec.plugin;
+  const { data: plugin, isLoading, error } = usePlugin('Panel', kind);
+
+  if (error) {
+    return <ErrorAlert error={error} />;
+  }
+
+  // TODO: Proper loading indicator
+  if (isLoading) {
+    return null;
+  }
+
+  if (plugin === undefined) {
+    throw new Error(`Missing implementation for panel plugin with kind '${kind}'`);
+  }
+
+  const { panelOptionsEditorComponents, hideQueryEditor } = plugin as PanelPlugin;
+  let tabs: OptionsEditorTabsProps['tabs'] = [];
+
+  if (!hideQueryEditor && panelDefinition.spec.queries !== undefined) {
```

Comment:
If a user accidentally forgets to add `queries` property, is the editor hidden? I think it should just create an empty `[]` in that case. Since people edit dashboard config by hand, there will be use cases like this.

Reply:
good call, updated this so we only hide query editor if `hideQueryEditor` is true